cranelift/codegen/src/isa/x64/lower.rs

```diff
@@ -1723,7 +1723,11 @@ fn lower_insn_to_regs<C: LowerCtx<I = Inst>>(
                                     8,
                                     OperandSize::Size32,
                                 ));
-                                ctx.emit(Inst::xmm_mov(SseOpcode::Pmovsxbw, RegMem::reg(rhs), dst));
+                                ctx.emit(Inst::xmm_mov(
+                                    SseOpcode::Pmovsxbw,
+                                    RegMem::reg(dst.to_reg()),
+                                    dst,
+                                ));
                                 ctx.emit(Inst::xmm_rm_r(
                                     SseOpcode::Pmullw,
                                     RegMem::reg(tmp_reg.to_reg()),
```

Comment:
This may be my lack of knowledge about the x64 backend, but the above instruction

> `Inst::gen_move(tmp_reg, lhs, output_ty)`

has the destination as the first parameter (`tmp_reg` I believe) and the parameters to the instruction as the second (`lhs`).

Does that mean that this should be `dst` as the first parameter since that should be the destination? (or does `xmm_rm_r` behave differently in that the destination is the second parameter?)

Reply:
Hm although given `xmm_rm_r_imm` it looks like the destination isn't the first argument, so I may just not know the backend here in this regard.

Reply:
> This may be my lack of knowledge about the x64 backend, but the above instruction
> 
> > `Inst::gen_move(tmp_reg, lhs, output_ty)`
> 
> has the destination as the first parameter (`tmp_reg` I believe) and the parameters to the instruction as the second (`lhs`).
> 
> Does that mean that this should be `dst` as the first parameter since that should be the destination? (or does `xmm_rm_r` behave differently in that the destination is the second parameter?)

Yes .. good observation. gen_move is unique in this way by having the first operand as the destination. Not sure the history behind that but it is something that perhaps should be refactored to avoid confusion.